app/commands/update_order.rb

```diff
@@ -0,0 +1,105 @@
+class UpdateOrder < Rectify::Command
+  def initialize form, names
+    @form = form
+    @names = names
+    @names = [names] if names.class == Symbol
+  end
+
+  def call
+    return broadcast(:invalid) if @form.invalid?
+
+    transaction do
+      @names.each do |name|
+        send "set_or_update_#{name}"
+      end
+    end
+
+    broadcast(:ok)
+  end
+
+  private
+    def set_or_update_order_items
+      order_items.keys.each do |id|
+        order_item = order_item id
+        if order_item_params(id)['quantity'].to_i > order_item.book.instock
+          order_item.update(quantity: order_item.book.instock)
+        elsif order_item_params(id)['quantity'].to_i < 1
+          order_item.update(quantity: 1)
+        else
+          order_item.update(order_item_params(id))
+        end
+      end if order_items
+    end
+
+    def order_items
+      @form.order_items
+    end
+
+    def order_item_params id
+      order_items.require(id).permit('quantity')
+    end
+
+    def order_item id
+      OrderItem.find(id.to_i)
+    end
+
+    def set_or_update_coupon
+      current_order.update(coupon: Coupon.find_by(code: coupon))
+    end
+
+    def coupon
+      @form.coupon
+    end
+
+    def set_or_update_addresses
+      billing_address = @form.billing_address.id == '' ? create_address(:billing_address) : 
+        update_address(:billing_address)
```

Comment:
1. @form.billing_address.id.present?
2. if or move condition to separate method